package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +189,34 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.popen3(path, ...)
+	local r0, w0 = posix_unistd.pipe()
+	local r1, w1 = posix_unistd.pipe()
+	local r2, w2 = posix_unistd.pipe()
+
+	local pid = posix_unistd.fork()
+
+	if pid < 0 then
+		return nil
+	elseif pid == 0 then
+		posix_unistd.close(w0)
+		posix_unistd.close(r1)
+		posix_unistd.close(r2)
+
+		posix_unistd.dup2(r0, posix_unistd.STDIN_FILENO)
+		posix_unistd.dup2(w1, posix_unistd.STDOUT_FILENO)
+		posix_unistd.dup2(w2, posix_unistd.STDERR_FILENO)
+
+		posix_unistd.execp(path, ...)
+		return nil
```

Comment:
You are in a forked process here. When the exec fails, the child process must be terminated without running any cleanup (as that would confuse the parent process). `unistd._exit(127)` takes care of this (with 127 being a common exit code that appears in other similar code)

Reply:
reoslved in 246ed9760184da60ff3b3908afbcce2532cb4578